ArtOfIllusion/src/artofillusion/image/ExternalImage.java

```diff
@@ -0,0 +1,363 @@
+/* Copyright (C) 2017 by Petri Ihalainen
+
+   This program is free software; you can redistribute it and/or modify it under the
+   terms of the GNU General Public License as published by the Free Software
+   Foundation; either version 2 of the License, or (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful, but WITHOUT ANY 
+   WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A 
+   PARTICULAR PURPOSE.  See the GNU General Public License for more details. */
+
+package artofillusion.image;
+
+import artofillusion.*;
+import artofillusion.math.*;
+import buoy.widget.BStandardDialog;
+import java.awt.*;
+import java.awt.image.*;
+import java.io.*;
+import java.util.Date;
+import javax.imageio.*;
+
+public class ExternalImage extends ImageMap
+{
+  private File externalFile;
+  private String loadingError;
+  private ImageMap imageMap;
+  private int w, h;
+  private String type, lastAbsolutePath, lastRelativePath;
+  private File imageFile;
+  private Image brokenImage;
+  private boolean connected, newLink;
+
+  // ** Create an external image out of a image file */
+  
+  public ExternalImage(String path) throws InterruptedException
+  {
+    this(new File(path));
+  }
+
+  // ** Create an external image out of an image file */
+
+  public ExternalImage(File file) throws InterruptedException
+  {
+    newLink = true;
+    try
+    {
+      loadExternalImage(file);
+    }
+    catch(Exception e)
+    {
+      throw new InterruptedException();
+    }
+    imageFile = file;
+    setDataCreated(file);
+    lastAbsolutePath = file.getAbsolutePath();
+    // lastRelativePath can not be created without the path to the scene.
+  }
+
+  private Image loadIcon(String iconName)
+  {  
+    try 
+    {
+      return ImageIO.read(ExternalImage.class.getResource("/artofillusion/image/icons/" + iconName));
```

Comment:
Instead of doing this, it's better to load icons through `ThemeManager`.  That way they can be customized by themes.